core/blockchain.go

```diff
@@ -293,6 +293,16 @@ func IsEpochBlock(block *types.Block) bool {
 	return shard.Schedule.IsLastBlock(block.NumberU64() - 1)
 }
 
+// IsEpochBlockByNumber returns whether this block is the first block of an epoch.
+// by checking the block number
+func IsEpochBlockByNumber(blockNum uint64) bool {
```

Comment:
Daniel add a utility for this purpose. IsLastBlockInEpoch(). that's the correct way to check. Because only beacon chain follow fixed epoch schedule. Other chains follow beacon chain's epoch change timing.

Reply:
let me check.

Reply:
changed.